### ISSUE TITLE
Some current test failures on Windows

### ISSUE THREAD

**Issue by djberg96** [ManageIQ/azure-armrest#332]
Windows 8
ruby 2.4.2p198 (2017-09-14 revision 59899) [i386-mingw32]

While not a high priority, this library should be platform neutral, so we shouldn't be seeing failures on any platform. I suspect this is just the way we've written our tests rather than a core library issue, but thought I should post.

```
Failures:

  1) Azure::Armrest::Configuration instances tokens token generation caches the token to be reused for the same cl
     Failure/Error: expect(subject.token).to eql(token)

       expected: "Bearer eyJ0eXAiOiJKV1Q"
            got: "xxx"

       (compared using eql?)
     # ./spec/configuration_spec.rb:185:in `block (5 levels) in <top (required)>'

  2) BaseModel inspection methods defines a pretty_print method when pp is available
     Failure/Error: expect(base.pretty_inspect).to match(expected)

       expected "#<Azure::Armrest::BaseModel:0x03b95f98 name=\"test\", array=[\"stuff\"], age=33>\n" to match /\A#
::Armrest::BaseModel:0x\h+\n/
       Diff:
       @@ -1,2 +1,2 @@
       -/\A#<Azure::Armrest::BaseModel:0x\h+\n/
       +#<Azure::Armrest::BaseModel:0x03b95f98 name="test", array=["stuff"], age=33>

     # ./spec/models/base_model_spec.rb:146:in `block (3 levels) in <top (required)>'

  3) BaseModel #attr_from_hash maps it source location to the correct file
     Failure/Error: expect(subject.method(:first_name_from_hash).source_location).to include(__FILE__)
       expected ["C", 0] to include "C:/Users/Daniel/Dev/azure-armrest-djberg96/spec/models/base_model_spec.rb"
     # ./spec/models/base_model_spec.rb:291:in `block (3 levels) in <top (required)>'
     # ./spec/models/base_model_spec.rb:279:in `block (3 levels) in <top (required)>'

  4) BaseModel #attr_from_hash with multiple attributes maps it source location of each method to the correct file
     Failure/Error: expect(subject.method(:first_name_from_hash).source_location).to include(__FILE__)
       expected ["C", 0] to include "C:/Users/Daniel/Dev/azure-armrest-djberg96/spec/models/base_model_spec.rb"
     # ./spec/models/base_model_spec.rb:313:in `block (4 levels) in <top (required)>'
     # ./spec/models/base_model_spec.rb:301:in `block (4 levels) in <top (required)>'
     # ./spec/models/base_model_spec.rb:279:in `block (3 levels) in <top (required)>'

  5) BaseModel #attr_from_hash with nested attributes maps it source location to the correct file
     Failure/Error: expect(subject.method(:street_address_from_hash).source_location).to include(__FILE__)
       expected ["C", 0] to include "C:/Users/Daniel/Dev/azure-armrest-djberg96/spec/models/base_model_spec.rb"
     # ./spec/models/base_model_spec.rb:335:in `block (4 levels) in <top (required)>'
     # ./spec/models/base_model_spec.rb:324:in `block (4 levels) in <top (required)>'
     # ./spec/models/base_model_spec.rb:279:in `block (3 levels) in <top (required)>'
```

@NickLaMuro I think 3, 4 and 5 are yours. :)

**Comment by NickLaMuro** [comment]
@djberg96 Should we set up running tests on appveyor then?  https://www.appveyor.com/

Might take a look at this later today, though, if I get "really bored", but if you figure it out, cool.

**Comment by djberg96** [comment]
@NickLaMuro Looks like it's this line:

https://github.com/ManageIQ/azure-armrest/blob/master/lib/azure/armrest/model/base_model.rb#L59

Since you're splitting on ":", it's getting messed up because a Windows path is like "C:/foo/bar".

I thought there was a nicer way to do this in Ruby 2.x, but I haven't looked in a while.

**Comment by NickLaMuro** [comment]
@djberg96 Oh yeah, makes sense.  Hmm... I will take a look into this while waiting on some of my performance script runs to complete.  Seems like this would be a solved problem.

**Comment by NickLaMuro** [comment]
@djberg96 So yeah, the fix in #344 is arguably worth a release as stacktraces including these meta-programmed methods will not be correct, but depends on if you think that would come up frequently or not.

Guess it doesn't hurt waiting for it to be released when the next one is scheduled, though, as it shouldn't break things, just confuse folks when debugging.

Thanks for finding the bug for me, by the way.  Was very simple to find the right method to use for it after that.

**Comment by djberg96** [comment]
@NickLaMuro I'll include it as part of the next release, but I'm not going to rush one out just for this. Anyway, thanks!
